mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraPerUserMaxQuotaDao.java

```diff
@@ -0,0 +1,188 @@
+    private final PreparedStatement setDefaultMaxMessageStatement;
+    private final PreparedStatement getDefaultMaxStatement;
+    private final PreparedStatement removeMaxStorageStatement;
+    private final PreparedStatement removeMaxMessageStatement;
+    private final PreparedStatement removeDefaultMaxQuotaStatement;
+
+    @Inject
+    public CassandraPerUserMaxQuotaDao(Session session) {
+        this.session = session;
+        this.setMaxStorageStatement = session.prepare(insertInto(CassandraMaxQuota.TABLE_NAME)
+            .value(CassandraMaxQuota.QUOTA_ROOT, bindMarker())
+            .value(CassandraMaxQuota.STORAGE, bindMarker()));
+        this.setMaxMessageStatement = session.prepare(insertInto(CassandraMaxQuota.TABLE_NAME)
+            .value(CassandraMaxQuota.QUOTA_ROOT, bindMarker())
+            .value(CassandraMaxQuota.MESSAGE_COUNT, bindMarker()));
+        this.getMaxStorageStatement = session.prepare(select(CassandraMaxQuota.STORAGE)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+        this.getMaxMessageStatement = session.prepare(select(CassandraMaxQuota.MESSAGE_COUNT)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+        this.getDefaultMaxStatement = session.prepare(select(CassandraDefaultMaxQuota.VALUE)
+            .from(CassandraDefaultMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDefaultMaxQuota.TYPE, bindMarker(CassandraDefaultMaxQuota.TYPE))));
+        this.setDefaultMaxMessageStatement = session.prepare(insertInto(CassandraDefaultMaxQuota.TABLE_NAME)
+            .value(CassandraDefaultMaxQuota.TYPE, CassandraDefaultMaxQuota.MESSAGE)
+            .value(CassandraDefaultMaxQuota.VALUE, bindMarker()));
+        this.setDefaultMaxStorageStatement = session.prepare(insertInto(CassandraDefaultMaxQuota.TABLE_NAME)
+            .value(CassandraDefaultMaxQuota.TYPE, CassandraDefaultMaxQuota.STORAGE)
+            .value(CassandraDefaultMaxQuota.VALUE, bindMarker()));
+        this.removeDefaultMaxQuotaStatement = session.prepare(delete().all()
+            .from(CassandraDefaultMaxQuota.TABLE_NAME)
+            .where(eq(CassandraDefaultMaxQuota.TYPE, bindMarker(CassandraDefaultMaxQuota.TYPE))));
+        this.removeMaxStorageStatement = session.prepare(delete().column(CassandraMaxQuota.STORAGE)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+        this.removeMaxMessageStatement = session.prepare(delete().column(CassandraMaxQuota.MESSAGE_COUNT)
+            .from(CassandraMaxQuota.TABLE_NAME)
+            .where(eq(CassandraMaxQuota.QUOTA_ROOT, bindMarker())));
+    }
+
+    public void setMaxStorage(QuotaRoot quotaRoot, QuotaSize maxStorageQuota) {
+        session.execute(setMaxStorageStatement.bind(quotaRoot.getValue(), quotaValueToLong(maxStorageQuota)));
```

Comment:
Why don't we use the CassandraAsyncExecutor like everywhere else?

Also this DAO can return CompletableFuture, like other DAOs...

Reply:
Because I didn't took care of that, I just made consistent change base on current state